src/Adapter/Order/OrderAmountUpdater.php

```diff
@@ -107,13 +107,29 @@ public function update(
         $order->total_products = (float) $cart->getOrderTotal(false, Cart::ONLY_PRODUCTS, $orderProducts, $carrierId);
         $order->total_products_wt = (float) $cart->getOrderTotal(true, Cart::ONLY_PRODUCTS, $orderProducts, $carrierId);
 
+        $order->total_wrapping = abs($cart->getOrderTotal(true, Cart::ONLY_WRAPPING, $orderProducts, $carrierId));
+        $order->total_wrapping_tax_excl = abs($cart->getOrderTotal(false, Cart::ONLY_WRAPPING, $orderProducts, $carrierId));
+        $order->total_wrapping_tax_incl = abs($cart->getOrderTotal(true, Cart::ONLY_WRAPPING, $orderProducts, $carrierId));
+
+        $totalShippingTaxIncluded = $order->total_shipping_tax_incl;
+        $totalShippingTaxExcluded = $order->total_shipping_tax_excl;
+
         $order->total_shipping = $cart->getOrderTotal(true, Cart::ONLY_SHIPPING, $orderProducts, $carrierId);
         $order->total_shipping_tax_excl = $cart->getOrderTotal(false, Cart::ONLY_SHIPPING, $orderProducts, $carrierId);
         $order->total_shipping_tax_incl = $cart->getOrderTotal(true, Cart::ONLY_SHIPPING, $orderProducts, $carrierId);
 
-        $order->total_wrapping = abs($cart->getOrderTotal(true, Cart::ONLY_WRAPPING, $orderProducts, $carrierId));
-        $order->total_wrapping_tax_excl = abs($cart->getOrderTotal(false, Cart::ONLY_WRAPPING, $orderProducts, $carrierId));
-        $order->total_wrapping_tax_incl = abs($cart->getOrderTotal(true, Cart::ONLY_WRAPPING, $orderProducts, $carrierId));
+        if (!$this->configuration->get('PS_ORDER_RECALCULATE_SHIPPING')) {
+            $shippingDiffTaxIncluded = $order->total_shipping_tax_incl - $totalShippingTaxIncluded;
```

Comment:
but $totalShippingTaxIncluded == $order->total_shipping_tax_incl

So $totalShippingTaxIncluded == 0

Cant we just not modify total_paid, total_paid_tax_incl and total_paid_tax_excl ?

Reply:
@sowbiba I'm not sure to understand but, `$order->total_shipping_tax_incl` is not 0 if shipping wasn't free 🤔 
Also, `total_paid`, `total_paid_tax_incl` & `total_paid_tax_excl` should be the result of shipping + products total (+ fees) so we cannot update one without updating the other.

Reply:
On line 114, we have
$totalShippingTaxIncluded = $order->total_shipping_tax_incl;
Line 122
$shippingDiffTaxIncluded = $order->total_shipping_tax_incl - $totalShippingTaxIncluded;

So $shippingDiffTaxIncluded == 0

Reply:
Ok, you are right, I just ignored the 3 line between these statements :D :D